src/ast/ast_validations.c

```diff
@@ -121,6 +121,9 @@ static void _AST_GetWithReferences(const cypher_astnode_t *node, rax *identifier
 		const cypher_astnode_t *child = cypher_ast_with_get_projection(node, i);
 		_AST_GetIdentifiers(child, identifiers);
 	}
+
+	const cypher_astnode_t *order_by = cypher_ast_with_get_order_by(node);
+	if(order_by) _AST_GetIdentifiers(order_by, identifiers);
```

Comment:
Should the same apply to `RETURN` clause ?

Reply:
`RETURN` already behaves as expected; `WITH` is unique because it has separate handling for inputs and outputs and the `ORDER BY` subclause was previously overlooked.